ddtrace/contrib/grpc/patch.py

```diff
@@ -0,0 +1,48 @@
+import grpc
+import wrapt
+
+from ddtrace import Pin
+from ...utils.wrappers import unwrap
+
+from .client_interceptor import create
+
+def patch():
+    # patch only once
+    if getattr(grpc, '__datadog_patch', False):
+        return
+    setattr(grpc, '__datadog_patch', True)
+    Pin(service='grpc', app='grpc', app_type="grpc").onto(grpc)
+
+    _w = wrapt.wrap_function_wrapper
+
+    _w('grpc', 'insecure_channel', _insecure_channel_with_interceptor)
+    _w('grpc', 'secure_channel', _secure_channel_with_interceptor)
+
+def unpatch():
+    if not getattr(grpc, '__datadog_patch', False):
+        return
+    setattr(grpc, '__datadog_patch', False)
+    unwrap(grpc, 'secure_channel')
+    unwrap(grpc, 'insecure_channel')
+
+def _insecure_channel_with_interceptor(wrapped, instance, args, kwargs):
+    channel = wrapped(*args, **kwargs)
+    target = args[0]
+    (host, port) = get_host_port(target)
+    channel = _intercept_channel(channel, host, port)
+    return channel
+
+def _secure_channel_with_interceptor(wrapped, instance, args, kwargs):
+    channel = wrapped(*args, **kwargs)
```

Comment:
We will want to add the pin check here.

```python
pin = Pin.get_from(instance)
if not pin or not pin.enabled():
    return wrapped(*args, **kwargs)
```

Reply:
And then we can pass this pin into `GrpcClientInterceptor` as a constructor argument rather than needing to pull via `Pin.get_from(grpc)`.

This also will allow someone to override the pin on a specific connection.